infra/bridge/src/controllers/identity.js

```diff
@@ -0,0 +1,167 @@
+'use strict'
+
+const express = require('express')
+const get = require('lodash/get')
+
+const router = express.Router()
+const { GrowthEvent } = require('@origin/growth-event/src/resources/event')
+const {
+  loadIdentityAttestationsMetadata,
+  loadIdentityAddresses,
+  recordGrowthProfileEvent,
+  recordGrowthAttestationEvents
+} = require('@origin/identity/src/utils')
+const Identity = require('@origin/identity/src/models').Identity
+const validator = require('@origin/validator')
+
+const { pinIdentityToIpfs, postToEmailWebhook } = require('../utils/identity')
+const {
+  identityReadVerify,
+  identityWriteVerify
+} = require('../utils/validation')
+
+const logger = require('../logger')
+
+/**
+ * Loads the identity associated with an eth address.
+ * Return status:
+ *  - 200 if an identity was found in the database.
+ *  - 204 if no identity found.
+ *  - 500 if an identity row was found but with empty data in the database.
+ *
+ *  @args {string} req.ethAddress Eth address of the user.
+ *  @returns {{identity: Object, ipfsHash: string}}
+ */
+router.get('/', identityReadVerify, async (req, res) => {
+  const ethAddress = req.query.ethAddress.toLowerCase()
+  logger.debug(`identity/read called for addr ${ethAddress}`)
+
+  // Lookup the identity associated with the eth address.
+  const identity = await Identity.findOne({ where: { ethAddress } })
+
+  if (!identity || !identity.data) {
+    logger.debug(`No identity found for eth address ${ethAddress}.`)
+    return res.status(204).end()
+  }
+
+  // Check validity of the data before returning it.
+  const errors = []
+  for (const field of ['identity', 'ipfsHash', 'ipfsHashHistory']) {
+    if (!get(identity.data, field)) {
+      errors.push(`Identity ${ethAddress} is missing field data.${field}`)
+    }
+  }
+  if (errors.length) {
+    logger.error(errors)
+    return res.status(500).send({ errors })
+  }
+
+  return res.status(200).send({
+    identity: identity.data.identity,
+    ipfsHash: identity.data.ipfsHash
+  })
+})
+
+/**
+ * Writes an identity to the database.
+ *
+ * TODO(franck): Authenticate the request.
+ *
+ * @args {string} req.query.ethAddress: Eth address of the user.
+ * @args {Object} req.body.identity: Identity JSON blob.
+ * @args {string} req.body.ipfsHash: IPFS hash of the identity JSON blob.
+ * @returns {{id: string}} Returns the eth addresses passed as input.
+ */
+router.post('/', identityWriteVerify, async (req, res) => {
+  const ethAddress = req.query.ethAddress.toLowerCase()
```

Comment:
Same comment as above, What happens if `ethAddress` is a proxy address?

Reply:
Done